src/dxtbx/format/FormatNXmxDLS.py

```diff
@@ -13,36 +12,63 @@
 from dxtbx.format.FormatNXmx import FormatNXmx
 
 
-def get_bit_depth_from_meta(meta_file_name):
+def get_bit_depth_from_meta(meta_file_name: Path) -> int:
     with h5py.File(meta_file_name) as f:
-        return int(f["/_dectris/bit_depth_image"][()])
+        with contextlib.suppress(KeyError):
+            return int(f["/_dectris/bit_depth_image"][()])
 
+        # This might be a very old meta file, that only had 'datatype'?
+        if "datatype" in f:
+            unique_vals = np.unique(f["datatype"])
+            if not len(unique_vals) == 1:
+                raise RuntimeError(
+                    f"Error: Could not determine bit depth from metafile {meta_file_name} (multiple bit depths)"
+                )
+            return np.dtype(unique_vals[0]).itemsize * 8
 
-def find_meta_filename(master_like: Union[str, Path]) -> Union[str, Path]:
+    raise RuntimeError("Cannot determine metafile bit depth")
+
+
+def find_meta_filename(master_like: Path) -> Path:
     """
     Find the path to the '..._meta.h5' file in the same directory as the master file.
     Args:
         master_like:  File path of the master HDF5 file.
+
     Returns:
         File path of the HDF5 metadata '..._meta.h5' file.
+
+    Raises:
+        RuntimeError: If no meta-file was found.
     """
 
-    def _local_visit(name):
+    def _local_visit(name) -> Path | None:
         obj = f[name]
         if not hasattr(obj, "keys"):
             return None
         for k in obj.keys():
             kclass = obj.get(k, getlink=True, getclass=True)
             if kclass is h5py._hl.group.ExternalLink:
-                kfile = obj.get(k, getlink=True).filename
-                if kfile.split(".")[0].endswith("meta"):
+                kfile = master_like.parent / obj.get(k, getlink=True).filename
+                if kfile.stem.endswith("meta"):
                     return kfile
+        return None
 
-    master_dir = os.path.split(master_like)[0]
+    master_like = Path(master_like)
     with h5py.File(master_like) as f:
         meta_filename = f.visit(_local_visit)
 
-    return os.path.join(master_dir, meta_filename)
+    if meta_filename is None:
+        # Try a fallback of looking for a file named the same but "meta.h5"
+        look_for = master_like.with_name(
+            master_like.stem.removesuffix("_master") + "_meta.h5"
```

Comment:
i.e. if `foo.nxs` is passed what does this do?

Reply:
```python
>>> Path("foo.nxs").stem.removesuffix("_master") + "_meta.h5"
'foo_meta.h5'
>>> Path("foo_master.h5").stem.removesuffix("_master") + "_meta.h5"
'foo_meta.h5'
```